### PR TITLE
feat: add renderJavascript option

### LINKED ISSUE #78: Execute Javascript on webpage
Some Netlify websites requires javascript to be fully executed before we extract information.
Even though it's usually recommended to have SSR and/or static webpage, 
you can easily plug any API to load the content in the frontend; making the crawler loose some info.

The crawler does support javascript rendering but it's not exposed in the plugin.
Should we do it?

### PR DESCRIPTION
**Pull request by bodinsamuel**
Closes #78

Add `renderJavascript` option to enable javascript rendering in a browser

### CODE REVIEW

**Comment by bodinsamuel** [comment]
Tested locally with running crawler and `yarn dev`

File: plugin/README.md [hunk 1]
```diff
@@ -28,6 +28,8 @@ Plugin inputs can be set in `netlify.toml`. They're all optional.
   Putting "pathPrefix: /blog" will alias `/blog` to `/`
 - `customDomain` - The custom domain that you use, if it's not possible to define it on your Netlify's settings.
   Putting "customDomain: example.com" will alias `example.com` to `<your-site-url>.netlify.app`
+- `renderJavascript` - _Default: `false`_ Use to enable the javascript rendering of your website, useful for SPA.
+  See [documentation](https://www.algolia.com/doc/api-reference/crawler/configuration/render-java-script/) to understood the implication of this option.
```
Comment (sbellone) [inline plugin/README.md:32]:
```suggestion
  See [documentation](https://www.algolia.com/doc/api-reference/crawler/configuration/render-java-script/) to understand the implication of this option.
```

File: plugin/README.md [hunk 2]
```diff
@@ -28,6 +28,8 @@ Plugin inputs can be set in `netlify.toml`. They're all optional.
   Putting "pathPrefix: /blog" will alias `/blog` to `/`
 - `customDomain` - The custom domain that you use, if it's not possible to define it on your Netlify's settings.
   Putting "customDomain: example.com" will alias `example.com` to `<your-site-url>.netlify.app`
+- `renderJavascript` - _Default: `false`_ Use to enable the javascript rendering of your website, useful for SPA.
```
Comment (sbellone) [inline plugin/README.md:31]:
```suggestion
- `renderJavascript` - _Default: `false`_ If true, we will use JavaScript to render your website. Useful for Single Page Applications.
```

File: docs/Extraction.md [hunk 3]
```diff
@@ -199,8 +199,8 @@ Learn more about [canonical meta tag](https://support.google.com/webmasters/answ
 
 ## Executing Javascript
 
-The Algolia Crawler supports JavaScript-rendered websites, but we don't yet expose this option in the plugin.
-Please [contact us](mailto:support@algolia.com) to enable it.
+If your website needs it, for example if you have a React application without Server Side Rendering,
+you can enable Javascript rendering by updating the option in your [netlify.toml](/plugin/README.md#inputs)
```
Comment (sbellone) [inline docs/Extraction.md:203]:
```suggestion
you can enable Javascript rendering by setting the `renderJavascript` option in your [netlify.toml](/plugin/README.md#inputs)
```

File: docs/Extraction.md [hunk 4]
```diff
@@ -199,8 +199,8 @@ Learn more about [canonical meta tag](https://support.google.com/webmasters/answ
 
 ## Executing Javascript
 
-The Algolia Crawler supports JavaScript-rendered websites, but we don't yet expose this option in the plugin.
-Please [contact us](mailto:support@algolia.com) to enable it.
+If your website needs it, for example if you have a React application without Server Side Rendering,
```
Comment (sbellone) [inline docs/Extraction.md:202]:
```suggestion
If your website needs it, for example if you have a React application without server-side rendering,
```

File: plugin/manifest.yml [hunk 5]
```diff
@@ -12,3 +12,5 @@ inputs:
     description: The prefix of your website if it's not at the root level, e.g /blog
   - name: customDomain
     description: The custom domain that you use, if it's not possible to define it on your Netlify's settings.
+  - name: renderJavascript
+    description: Use to enable the javascript rendering of your website.
```
Comment (sbellone) [inline plugin/manifest.yml:16]:
```suggestion
    description: If true, we will use JavaScript to render your website.
```

**Review by sbellone — COMMENTED** [review]
Can we use the same camel cased name that we have in the Crawler configuration? `renderJavaScript`. Since we are referring to our doc, it might confuse users (and actually, it will confuse us too in the future).

**Comment by bodinsamuel** [comment]
> Can we use the same camel cased name that we have in the Crawler configuration? renderJavaScript.

Rha this naming 💀 

I need to update the API then

**Comment by sbellone** [comment]
> Rha this naming 💀

It's the official naming 🤷 :)

File: plugin/README.md [hunk 6]
```diff
@@ -28,6 +28,8 @@ Plugin inputs can be set in `netlify.toml`. They're all optional.
   Putting "pathPrefix: /blog" will alias `/blog` to `/`
 - `customDomain` - The custom domain that you use, if it's not possible to define it on your Netlify's settings.
   Putting "customDomain: example.com" will alias `example.com` to `<your-site-url>.netlify.app`
+- `renderJavaScript` - _Default: `false`_ If true, we will use JavaScript to render your website. Useful for Single Page Applications.
+  See [documentation](https://www.algolia.com/doc/api-reference/crawler/configuration/render-java-script/) to understand the implication of this option.
```
Comment (sbellone) [inline plugin/README.md:32]:
```suggestion
  See the [documentation](https://www.algolia.com/doc/api-reference/crawler/configuration/render-java-script/) to understand the implication of this option.
```

**Pull request merged**